### PR TITLE
Export environment checker and interpreter config to allow customization

### PR DESCRIPTION
**Pull request by turbolent**
## Description

The default checker and interpreter configurations may want to be changed, e.g. by the test framework. 
Export these two fields

______

- [x] Targeted PR against `master` branch
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work
- [x] Code follows the [standards mentioned here](https://github.com/onflow/cadence/blob/master/CONTRIBUTING.md#styleguides)
- [x] Updated relevant documentation 
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [x] Added appropriate labels

### CODE REVIEW

File: runtime/environment.go [hunk 1]
```diff
@@ -63,18 +63,23 @@ type Environment interface {
...
 type interpreterEnvironment struct {
-	config                                Config
-	baseActivation                        *interpreter.VariableActivation
-	baseValueActivation                   *sema.VariableActivation
-	runtimeInterface                      Interface
-	storage                               *Storage
-	coverageReport                        *CoverageReport
-	codesAndPrograms                      codesAndPrograms
+	config Config
+
```
Comment (dsainati1) [inline runtime/environment.go:67]: Can we remove the newlines here so the fields are all vertically aligned?

Comment (turbolent) [reply runtime/environment.go:67]: Hmm, intentionally added newlines to group related fields and hopped it would help

Comment (dsainati1) [reply runtime/environment.go:67]: I see. In that case I suppose it's fine but it is frustrating that grouping the fields this way also ruins the alignment.

Comment (turbolent) [reply runtime/environment.go:67]: yeah, I usually have whitespace changes disabled on GitHub for that reason

**Review by dsainati1 — APPROVED** [review]

**Review by SupunS — APPROVED** [review]
For the moment, I've added this locally: https://github.com/onflow/cadence/commit/a7c0cec13e950fcd7644f2b8e9b1453e57aeebde.

Should be OK to merge this 👍

**Pull request merged**